packages/web-react/src/components/Grid/__tests__/Grid.test.tsx

```diff
@@ -14,39 +14,114 @@ describe('Grid', () => {
   });
+
+  it('should render with custom spacing', () => {
+    const testId = 'grid-test-id';
+
+    render(<Grid spacing="space-600" data-testid={testId} />);
+
+    const element = screen.getByTestId(testId) as HTMLElement;
+
+    expect(element).toHaveStyle({ '--grid-spacing-x': 'var(--spirit-space-600)' });
+    expect(element).toHaveStyle({ '--grid-spacing-y': 'var(--spirit-space-600)' });
+  });
+
+  it('should render with custom spacing for each breakpoint', () => {
+    const testId = 'grid-test-id';
+
+    render(
+      <Grid spacing={{ mobile: 'space-100', tablet: 'space-1000', desktop: 'space-1200' }} data-testid={testId} />,
+    );
+
+    const element = screen.getByTestId(testId) as HTMLElement;
+
+    expect(element).toHaveStyle({ '--grid-spacing-x': 'var(--spirit-space-100)' });
+    expect(element).toHaveStyle({ '--grid-spacing-y': 'var(--spirit-space-100)' });
+    expect(element).toHaveStyle({ '--grid-spacing-x-tablet': 'var(--spirit-space-1000)' });
+    expect(element).toHaveStyle({ '--grid-spacing-y-tablet': 'var(--spirit-space-1000)' });
+    expect(element).toHaveStyle({ '--grid-spacing-x-desktop': 'var(--spirit-space-1200)' });
+    expect(element).toHaveStyle({ '--grid-spacing-y-desktop': 'var(--spirit-space-1200)' });
+  });
+
+  it('should render with custom vertical spacing for each breakpoint', () => {
+    const testId = 'grid-test-id';
+
+    render(
+      <Grid spacingY={{ mobile: 'space-100', tablet: 'space-1000', desktop: 'space-1200' }} data-testid={testId} />,
+    );
+
+    const element = screen.getByTestId(testId) as HTMLElement;
+
+    expect(element).toHaveStyle({ '--grid-spacing-y': 'var(--spirit-space-100)' });
+    expect(element).toHaveStyle({ '--grid-spacing-y-tablet': 'var(--spirit-space-1000)' });
+    expect(element).toHaveStyle({ '--grid-spacing-y-desktop': 'var(--spirit-space-1200)' });
+    expect(element).not.toHaveStyle({ '--grid-spacing-x': 'var(--spirit-space-100)' });
+    expect(element).not.toHaveStyle({ '--grid-spacing-x-tablet': 'var(--spirit-space-1000)' });
+    expect(element).not.toHaveStyle({ '--grid-spacing-x-desktop': 'var(--spirit-space-1200)' });
+  });
+
+  it('should render with custom spacing for only one breakpoint', () => {
+    const testId = 'grid-test-id';
+
+    render(<Grid spacing={{ tablet: 'space-1000' }} data-testid={testId} />);
+
+    const element = screen.getByTestId(testId) as HTMLElement;
+
+    expect(element).toHaveStyle({ '--grid-spacing-x-tablet': 'var(--spirit-space-1000)' });
+    expect(element).toHaveStyle({ '--grid-spacing-y-tablet': 'var(--spirit-space-1000)' });
+    expect(element).not.toHaveStyle({ '--grid-spacing-x': 'var(--spirit-space-100)' });
+    expect(element).not.toHaveStyle({ '--grid-spacing-y': 'var(--spirit-space-100)' });
+    expect(element).not.toHaveStyle({ '--grid-spacing-x-desktop': 'var(--spirit-space-1200)' });
+    expect(element).not.toHaveStyle({ '--grid-spacing-y-desktop': 'var(--spirit-space-1200)' });
```

Comment:
Well, in this test case, all these `not`s look quite random as neither `100` or `1200` are set in `render()` above 🙂.

Again, I think the right thing is to check `--grid-spacing-x` and `--grid-spacing-x-desktop` have the default value set.

Reply:
https://github.com/lmc-eu/spirit-design-system/pull/1561/files#r1698606633 :(